test/tls.c

```diff
@@ -44,17 +44,19 @@
 #define DATA_VAL2 0x0a0a0a0a
 
 NEWLIB_THREAD_LOCAL volatile int data_var = DATA_VAL;
-NEWLIB_THREAD_LOCAL volatile int bss_var;
 _Alignas(128) NEWLIB_THREAD_LOCAL volatile int overaligned_data_var = DATA_VAL2;
+NEWLIB_THREAD_LOCAL volatile int bss_var;
+_Alignas(128) NEWLIB_THREAD_LOCAL volatile int overaligned_bss_var;
```

Comment:
Can you use a different larger alignment (e.g. 256) to make this test more corner cases?

Reply:
Like this? afe378e174a9467a029fb906073cd3fe5575c1be

Reply:
I had to laugh when I saw this change -- I made exactly the same addition to the tests when you figured out that there was probably a bug here. Check out pr #412 as that has this fix, along with some additional testing (which caught the problem in _init_tls()).